### PR TITLE
Rust Global WebContent

### PR DESCRIPTION
**Pull request by sparkplug0025**

### CODE REVIEW

File: rust/psibase/src/serve_http.rs [hunk 1]
```diff
@@ -210,3 +212,76 @@ pub fn serve_graphiql(request: &HttpRequest) -> Option<HttpReply> {
...
     }
 }
+
+#[derive(Debug, Fracpack, PartialEq, Reflect, Serialize, Deserialize)]
+#[fracpack(fracpack_mod = "fracpack")]
+#[reflect(psibase_mod = "crate")]
+pub struct WebContentRow {
+    pub path: String,
+    pub content_type: String,
+    pub content: Vec<u8>,
```
Comment (tbfleming) [inline rust/psibase/src/serve_http.rs:222]: hex

File: rust/test_service_elections/src/lib.rs [hunk 2]
```diff
@@ -71,6 +72,31 @@ mod service {
         }
     }
 
+    // TODO: create a macro for external records
```
Comment (tbfleming) [inline rust/test_service_elections/src/lib.rs:75]: The macro could expand to something like the "Inside the Service Module" section of https://doc-sys.psibase.io/rust-service/tables.html#storing-structs-defined-elsewhere

File: rust/psibase/src/hex.rs [hunk 3]
```diff
@@ -4,12 +4,14 @@ use std::marker::PhantomData;
 use std::ops::Deref;
 use std::str::FromStr;
 
-trait ToHex:
+pub trait ToHex:
```
Comment (tbfleming) [inline rust/psibase/src/hex.rs:7]: This should remain a private implementation detail. The ToString (implicit from Display) and FromStr traits provide conversion.

**Review by tbfleming — APPROVED** [review]

File: rust/psibase/src/hex.rs (hunk 3 above)
Comment (sparkplug0025) [reply rust/psibase/src/hex.rs:7]: I did this to enable `Vec<u8>::to_hex()` for debugging purposes. But I will rollback...

**Pull request merged**